### DISCUSSION TITLE
Specifying font with Japanese or Chinese font names

### DISCUSSION THREAD

**Discussion by coffeesmell** [ImageMagick/ImageMagick#4027] [Help]
Does imagemagick has limits when it looks for font with Japanese font names?

When I execute the command `magick convert -list font`, I got the a list of fonts and one of them is this:
```
 Font: 小塚ゴシック-Pro-R-Regular
    family: 小塚ゴシック Pro R
    style: Normal
    stretch: Normal
    weight: 400
    glyphs: c:\windows\fonts\kozgopro-regular.otf
```

And then I used the command `convert -size 320x100 xc:lightblue -font 小塚ゴシック-Pro-R-Regular -pointsize 72 -fill blue  -annotate +25+70 'text２０２１テキストtext' annotate.gif`

But ImageMagick cannot recognize the font "小塚ゴシック-Pro-R-Regular"

Has anybody encounter this kind of problem?

**Comment by fmw42** [comment]
Is it entered in the type.xml file? If not, then you would need to use the path to the font file rather than the font name

**Reply by coffeesmell** [reply]
@fmw42  
It is not entered in type.xml

I tried to use the path of the font file and it works.
I also tried entering it directly in type.xml and it works.

Thanks for this tip.

I just wondered why it cannot recognize such font without explicitly adding it in type.xml....could this be an IM bug?

**Reply by fmw42** [reply]
Enter the information in the type.xml file. If on Unix-like system, you can run the script at https://legacy.imagemagick.org/Usage/scripts/imagick_type_gen which should enter most fonts into the type.xml file so that names can be used.
